File: app/src/main/java/tech/salroid/filmy/network_stuff/MyHurlStack.java
package tech.salroid.filmy.network_stuff;

import com.android.volley.toolbox.HurlStack;

import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.URL;

import tech.salroid.filmy.BuildConfig;
import tech.salroid.filmy.FilmyApplication;
import tech.salroid.filmy.R;


 /** Filmy Application for Android
 * Copyright (c) 2016 Ramankit Singh (http://github.com/webianks).
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
public class MyHurlStack extends HurlStack {

    @Override
    protected HttpURLConnection createConnection(URL url) throws IOException {

        String api_key = BuildConfig.TMDB_API_KEY;
        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
        connection.setRequestProperty("Content-Type", "application/json");
        connection.setRequestProperty("trakt-api-version", "2");
        //connection.setRequestProperty("trakt-api-key", FilmyApplication.getContext().getString(R.string.api_key));

        return connection;
    }
}

File: app/src/main/java/tech/salroid/filmy/networking/MyHurlStack.kt
package tech.salroid.filmy.networking

import com.android.volley.toolbox.HurlStack
import tech.salroid.filmy.BuildConfig
import java.io.IOException
import java.net.HttpURLConnection
import java.net.URL

class MyHurlStack : HurlStack() {

    @Throws(IOException::class)
    override fun createConnection(url: URL): HttpURLConnection {
        val apiKey = BuildConfig.TMDB_API_KEY
        val connection = url.openConnection() as HttpURLConnection
        connection.setRequestProperty("Content-Type", "application/json")
        connection.setRequestProperty("trakt-api-version", "2")
        //connection.setRequestProperty("trakt-api-key", FilmyApplication.getContext().getString(R.string.api_key));
        return connection
    }
}